File: Types/PickerRowView.m
//
//  PickerRowView.m
//  Types
//
//  Created by Andrew Clissold on 11/24/13.
//  Copyright (c) 2013 Andrew Clissold. All rights reserved.
//

#import "PickerRowView.h"

@interface PickerRowView()

@property (nonatomic, strong) UILabel *titleLabel;

@end

@implementation PickerRowView

const CGFloat kViewHeight = 200;
const CGFloat kViewWidth = 140;
const CGFloat kLabelHeight = 20;
const CGFloat kMarginSize = 10;

+ (CGFloat)viewHeight {
    return kViewWidth;
}

+ (CGFloat)viewWidth {
    return kViewWidth;
}

- (id)initWithTitle:(NSString *)title image:(UIImage *)image {
    self = [super initWithFrame:CGRectMake(0.0, 0.0, kViewWidth, kViewHeight)];
    if (self) {
		CGFloat yCoord = (self.bounds.size.height - kLabelHeight) / 2;

        _titleLabel = [[UILabel alloc] initWithFrame:
                       CGRectMake(kMarginSize + image.size.width + kMarginSize,
                                  yCoord,
                                  CGRectGetWidth(self.frame) - kMarginSize + image.size.width + kMarginSize,
                                  kLabelHeight)];
        self.titleLabel.text = title;
        self.titleLabel.backgroundColor = [UIColor clearColor];
        [self addSubview:self.titleLabel];

        yCoord = (self.bounds.size.height - image.size.height) / 2;
        UIImageView *imageView = [[UIImageView alloc] initWithFrame:
                                  CGRectMake(kMarginSize,
                                             yCoord,
                                             image.size.width,
                                             image.size.height)];
        imageView.image = image;
        [self addSubview:imageView];
    }
    return self;
}

// Enable accessibility for this view.
- (BOOL)isAccessibilityElement
{
	return YES;
}

// Return a string that describes this view.
- (NSString *)accessibilityLabel
{
	return self.titleLabel.text;
}

@end

File: Types/PickerRowView.swift
//
//  PickerRowView.swift
//  Types
//
//  Created by Andrew Clissold on 7/25/15.
//  Copyright (c) 2015 Andrew Clissold. All rights reserved.
//

import UIKit

class PickerRowView: UIView {

    private var titleLabel: UILabel!

    private let viewHeight: CGFloat = 200
    private let viewWidth: CGFloat = 140
    private let labelHeight: CGFloat = 20
    private let marginSize: CGFloat = 10

    init(title: String, image: UIImage) {
        super.init(frame: CGRect(x: 0, y: 0, width: viewWidth, height: viewHeight))

        var y = (bounds.size.height - labelHeight) / 2
        titleLabel = UILabel(frame: CGRect(
            x: marginSize + image.size.width,
            y: y,
            width: frame.size.width + image.size.width,
            height: labelHeight))
        titleLabel.text = title
        titleLabel.font = UIFont(name: "AvenirNext-Regular", size: 17)
        titleLabel.backgroundColor = UIColor.clearColor()
        addSubview(titleLabel)

        y = (bounds.size.height - image.size.height) / 2
        let imageView = UIImageView(frame: CGRect(x: marginSize, y: y, width: image.size.width, height: image.size.height))
        imageView.image = image
        addSubview(imageView)

        isAccessibilityElement = true
        accessibilityLabel = title
    }

    required init(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

}
